### ISSUE TITLE
clasp doesn't find scripts attached to spreadsheets

### ISSUE THREAD

**Issue by dbieber** [google/clasp#382]
## Expected Behavior
clasp list includes apps scripts attached to spreadsheets
clasp clone <scriptId> should work if the scriptId is a script associated with a spreadsheet 

## Actual Behavior
clasp list does not include apps scripts associated with spreadsheets
clasp clone gives the following error if I use a scriptId from a script associated with a spreadsheet:
"> Did you provide the correct scriptId?"

## Steps to Reproduce the Problem

  1. Create a Google Sheet
  1. Tools > Script Editor  -- create a script associated with the sheet
  1. run clasp clone <scriptId> where the scriptId comes from the newly created script

## Specifications

  - Node version (`node -v`): v6.11.2
  - Version (`clasp -v`): 1.6.3
  - OS (Mac/Linux/Windows): Mac

Not sure if this is a bug report or a feature request, but either way, let me know what you recommend; I definitely would like to use a version control system with my scripts + to edit them locally instead of solely in the web IDE.

**Comment by dbieber** [comment]
Oh I found this! https://github.com/google/clasp/issues/208#issuecomment-395608767
Thanks @ShowMeBillyJo!
As per that comment, you can get the proper scriptId of a bound script via `File -> Project properties -> Info -> Script ID` and clasp will work with that ID.

This issue can be marked as a duplicate of https://github.com/google/clasp/issues/208.

**Comment by grant** [comment]
Yeah, for some reason Sheets has different URLs for attached scripts.
The Drive API can't find "bound" scripts, needed for `clasp list`. Marking as dup.

**Comment by MatthewGitter** [comment]
As of 11/21/24 clasp can list bound projects along with standalone projects, and it can clone and pull bound projects as it can standalone projects. However, the proper permissions scopes must be selected during clasp login in the authentication flow in the browser on the page introducing the list with "Select what clasp – The Apps Script CLI can access":

clasp list requires "See information about your Google Drive files. [✓]".
clasp clone <script ID> (where you copy <script ID> from the GAS project's Project Settings page) as well as clasp pull and clasp push require "Create and update Google Apps Script projects. [✓]" (also ensure that the local dir does not contain a .clasp.json file already, as the clone operation needs to overwrite it but will be blocked by an existing one).
clasp push additionally requires activating Google Apps Script API at https://script.google.com/u/5/home/usersettings , which can take a few minutes to take effect for clasp to use it.
